nemo/core/connectors/save_restore_connector.py

```diff
@@ -465,6 +466,30 @@ def _inject_model_parallel_rank_for_ckpt(self, dirname, basename):
         model_weights = inject_model_parallel_rank(model_weights)
         return model_weights
 
+    def _preserve_frozen_model_artifacts(self, model, current_model_artifact_dir):
+        """
+        Moves artifacts from a gpt model restored within MegatronGPTPromptLearningModel class
+        to MegatronGPTPromptLearningModel's .nemo file. Keeps the uniq artifact names so they
+        match the names in the gpt config file that is also stored as an artifact. 
+
+        Args:
+            model: ModelPT object to be saved.
+            current_model_artifact_dir: The tempdir where the model's other artifacts are being
+                                        held temporarily before creating the tar file. 
+        """
+
+        # Importing here to avoid circular import errors
+        from nemo.collections.nlp.models.language_modeling import MegatronGPTPromptLearningModel
```

Comment:
No - this is just bad - Nemo core should not ever import any of it's subdomains. Subclass this in NLP domain only.

Reply:
Yes, please use the `NLPSaveRestoreConnector` for this logic.

Reply:
I'm also wondering if we can do this using `register_artifact` somehow

Reply:
Sure, I can put it in `NLPSaveRestoreConnector`. It is also doable to use `register_artifact` if I write a custom save restore connector that inherents from `NLPSaveRestoreConnector`.

I didn't make a custom class initially because the vocab and merge files' artifact paths would need to be updated in the GPT model config which is it self an artifact. And the way I did things above seemed simpler.